teps/0125-add-credential-filter-to-entrypoint-logger.md

```diff
@@ -0,0 +1,334 @@
+---
+status: proposed
+title: Add credential filter to entrypoint logger
+creation-date: '2022-10-27'
+last-updated: '2022-10-27'
+authors:
+- '@useurmind'
+---
+
+# TEP-0125: Add credential filter to entrypoint logger
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes and Caveats](#notes-and-caveats)
+- [Design Details](#design-details)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [User Experience](#user-experience)
+  - [Performance](#performance)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+  - [Test Plan](#test-plan)
+  - [Infrastructure Needed](#infrastructure-needed)
+  - [Upgrade and Migration Strategy](#upgrade-and-migration-strategy)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+Currently tekton pipeline steps are printing out any secrets in plaintext. This 
+can be intended output from shell scripts but also accidental output from e.g. 
+the `set -x` shell command.
+
+Both is a security issue as logs are often propagated to log aggregators from a 
+central daemon. This can leak secrets.
+
+Therefore the tekton entrypoint which is running all pipeline step commands should
+filter secrets automatically where possible. 
+
+## Motivation
+
+Automatically avoid leaking secret values/credentials into the output logs of
+tekton pipelines.
+
+### Goals
+
+Secrets are filtered correctly from the output log in a way that the developer 
+of the pipeline does not need to configure this.
+
+Especially output from the `set -x` command should also be filtered correctly.
+
+### Non-Goals
+
+We only target secret values that tekton can automatically redact. We do not want 
+to redact secrets that are dynamically retrieved during the pipeline run and can
```

Comment:
Does this mean if e.g. your task queries a kube cluster for some secrets and prints their contents this isn't a case we're trying to address? I think that's fine, just a bit confused about what this means.

Possibly another non goal would be trying to redact secrets in params-- I think we just encourage people to not put secrets into params.

Reply:
I see it similar.

When I for example create an Oauth Token during the run of  a step using some client credentials that are mounted into the pod, the client credentials will be redacted but the Oauth Token will NOT.

Regarding redacting parameters that could be another story.

Reply:
probably worth putting "redacting params" in non goals-- useful but more challenging, and out of scope

Reply:
I have added this to the non goal section.

Reply:
awesome thank you!